### PR TITLE
MGMT-13889: Add periodic e2e test for removing a spoke node using ZTP

### PR DESCRIPTION
**Pull request by carbonin**
If REMOVE_SPOKE_NODE is set when running the ztp workflow this new step
will run after the spoke cluster is installed.

The step finds a worker agent and removes it from the spoke cluster by
annotating and deleting the corresponding BMH.

The worker is drained and then removed from the spoke.
The script waits for 30 minutes for the node to be removed.

This job also needs to set REDFISH_EMULATOR_IGNORE_BOOT_DEVICE=False
to ensure ironic is allowed to set to boot order in order to correctly
handle deprovisioning the node. Without this valide the host would boot
back into the node rather than into the discovery image when
deprovisioning.

Built on https://github.com/openshift/release/pull/40383

https://issues.redhat.com/browse/MGMT-13889

### CODE REVIEW

**Comment by carbonin** [comment]
/pj-rehearse periodic-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-remove-node-periodic

**Comment by carbonin** [comment]
/pj-rehearse periodic-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-remove-node-periodic

**Comment by carbonin** [comment]
/pj-rehearse periodic-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-remove-node-periodic

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
/pj-rehearse periodic-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-remove-node pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
/pj-rehearse periodic-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-remove-node pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
/pj-rehearse periodic-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-remove-node pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp

**Comment by carbonin** [comment]
This should be ready for review @adriengentil can you take a look (or suggest someone who could)?

**Review by adriengentil — APPROVED** [review]

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 1]
```diff
@@ -1131,6 +1131,22 @@ tests:
         ISO_IMAGE_TYPE=minimal-iso
       PLATFORM: nutanix
     workflow: assisted-nutanix
+- as: e2e-ai-operator-ztp-remove-node
```
Comment (adriengentil) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:1134]:
```suggestion
- as: e2e-ai-operator-ztp-remove-node-periodic
```

Comment (adriengentil) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:1134]: also can you add yourself to bugmaster configuration to get ping when the job fails? https://github.com/openshift-assisted/bugmaster-configurations/blob/main/config/channel/assisted-deployment-ci/bug_master_configuration.yaml#L7

Comment (carbonin) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:1134]: I had this originally but the job is already prefixed with `periodic-` so it seemed redundant.

Comment (adriengentil) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:1134]: we have a mix of presubmits and periodics in this file, we cannot have the same job names even though they are different types of jobs. That's why we suffix all the periodics.

Comment (carbonin) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:1134]: Ah interesting. So you can't have a single job definition be used for both a periodic and persubmit? That seems odd.

**Comment by carbonin** [comment]
/pj-rehearse ack

Ran rehearsals before changing the job name

**Pull request merged**